x-pack/packages/kbn-inference-endpoint-ui-common/src/components/inference_services.tsx

```diff
@@ -0,0 +1,359 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import {
+  getFieldValidityAndErrorMessage,
+  UseField,
+  useFormContext,
+  useFormData,
+} from '@kbn/es-ui-shared-plugin/static/forms/hook_form_lib';
+import { fieldValidators } from '@kbn/es-ui-shared-plugin/static/forms/helpers';
+import {
+  EuiFieldText,
+  EuiFieldTextProps,
+  EuiFormControlLayout,
+  EuiFormRow,
+  EuiHorizontalRule,
+  EuiInputPopover,
+  EuiSpacer,
+  keys,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { ConnectorFormSchema } from '@kbn/triggers-actions-ui-plugin/public';
+
+import * as i18n from '../translations';
+import { Config, ConfigEntryView, InferenceProvider, Secrets } from '../types/types';
+import { SERVICE_PROVIDERS } from './providers/render_service_provider/service_provider';
+import { DEFAULT_TASK_TYPE, ServiceProviderKeys } from '../constants';
+import { SelectableProvider } from './providers/selectable';
+import { TaskTypeOption, generateInferenceEndpointId, getTaskTypeOptions } from '../utils/helpers';
+import { ConfigurationFormItems } from './configuration/configuration_form_items';
+import { AdditionalOptionsFields } from './additional_options_fields';
+import { ProviderSecretHiddenField } from './hidden_fields/provider_secret_hidden_field';
+import { ProviderConfigHiddenField } from './hidden_fields/provider_config_hidden_field';
+
+interface InferenceServicesProps {
+  providers: InferenceProvider[];
+}
+
+export const InferenceServices: React.FC<InferenceServicesProps> = ({ providers }) => {
+  const [isProviderPopoverOpen, setProviderPopoverOpen] = useState(false);
+  const [providerSchema, setProviderSchema] = useState<ConfigEntryView[]>([]);
+  const [taskTypeOptions, setTaskTypeOptions] = useState<TaskTypeOption[]>([]);
+  const [selectedTaskType, setSelectedTaskType] = useState<string>(DEFAULT_TASK_TYPE);
+  const [taskTypeFormFields] = useState<ConfigEntryView[]>([]);
```

Comment:
After cleaning up taskTypeFormFields in AdditionalOptionsFields, this state also could be removed as not used.